src/decode.c

```diff
@@ -145,9 +145,38 @@ PacketAlert *PacketAlertCreate(void)
     return pa_array;
 }
 
+void PacketAlertRecycle(PacketAlert *pa_array)
+{
+    if (pa_array != NULL) {
+        for (int i = 0; i < packet_alert_max; i++) {
+            if (pa_array[i].json_info.next != NULL) {
+                struct JsonInfoList *current_json = pa_array[i].json_info.next;
+                while (current_json) {
+                    struct JsonInfoList *next_json = current_json->next;
+                    SCFree(current_json);
+                    current_json = next_json;
+                }
+            }
+            pa_array[i].json_info.json_string = NULL;
+            pa_array[i].json_info.next = NULL;
+        }
+    }
+}
+
 void PacketAlertFree(PacketAlert *pa)
 {
     if (pa != NULL) {
+        for (int i = 0; i < packet_alert_max; i++) {
+            /* first item is not allocated so start at second one */
+            if (pa[i].json_info.next != NULL) {
+                struct JsonInfoList *current_json = pa[i].json_info.next;
```

Comment:
Q: This is a bit weird to read. Why is current the next? I see the comment above but could we improve this somehow?

Reply:
Indeed, I'v replaced current_ by allocated_.